From ed9fae66f68f33f1353f723d294985fb100df7ca Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 8 Dec 2010 14:28:54 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: a8e93126a6f10d0a14ba8407ec112b1b3a5e2e97
Subject: [PATCH 1935/2588] drm/i915/gtt: Clear the cachelines upon resume

Required for my pineview system to not barf after resuming.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit a8e93126a6f10d0a14ba8407ec112b1b3a5e2e97)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 71c2b0f..86673e7 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -35,6 +35,8 @@ void i915_gem_restore_gtt_mappings(struct drm_device *dev)
 	struct drm_i915_gem_object *obj;
 
 	list_for_each_entry(obj, &dev_priv->mm.gtt_list, gtt_list) {
+		i915_gem_clflush_object(obj);
+
 		if (dev_priv->mm.gtt->needs_dmar) {
 			BUG_ON(!obj->sg_list);
 
@@ -51,7 +53,6 @@ void i915_gem_restore_gtt_mappings(struct drm_device *dev)
 					       obj->agp_type);
 	}
 
-	/* Be paranoid and flush the chipset cache. */
 	intel_gtt_chipset_flush();
 }
 
-- 
1.7.6

